### ISSUE TITLE
Cmake 3.16 on Mac

### ISSUE THREAD

**Issue by aviks** [JuliaPackaging/BinaryBuilder.jl#741]
Could I request `cmake` version of at least 3.16 on the mac image? I need it to compile XGBoost with openmp on the mac. 

ref: https://github.com/dmlc/xgboost/pull/5146#issuecomment-568312706

**Comment by giordano** [comment]
We install CMake from Alpine Linux repositories.  The latest stable version of Alpine has CMake 3.15...  Also Viral tried building xgboost some weeks ago and [had the same problem](https://github.com/JuliaPackaging/Yggdrasil/pull/655#issuecomment-604600262) :disappointed:   For context, CMake 3.16 has been released last November...

**Comment by Keno** [comment]
Time to BinaryBuild cmake so we can put it int the rootfs ;).

**Comment by giordano** [comment]
Well, at least a native build _should_ be straightforward

**Comment by aviks** [comment]
> Viral tried building xgboost some weeks ago

Yes, I am trying to figure out options to move forward with that work.

> The latest stable version of Alpine has CMake 3.15

Their `edge` branch has 3.17.1, but I suppose that doesnt help us? Looks like a new stable will be released in a couple of months.

**Comment by giordano** [comment]
https://github.com/JuliaPackaging/Yggdrasil/pull/853 is a proof of concept of building CMake 3.17.1 within the build environment.

What goes into the rootfs is decided by `0_RootFS/Rootfs/build_tarballs.jl`, we can either add the step to build CMake or use this CMake JLL package as dependency.  This second option a bit tricky, as the dependency would be installed in the "wrong" place and we would need all of its content consistently.  Maybe we can use the generated tarball as source, that would be easier and would save us CMake compilation time.
